src/sensai/vector_model.py

```diff
@@ -164,10 +248,10 @@ def predict(self, x: pd.DataFrame) -> pd.DataFrame:
         """
         x = self._computeInputs(x)
         y = self._predict(x)
+        # TODO or not TODO: why do we do this? Isn't it the implementation's responsibility to get it right?
```

Comment:
@opcode81 What do you think? If we remove this line, we could save some ram (negligible in most cases, of course) by always redeclaring x instead of having x and y.

Reply:
Currently, implementations don't correctly set the index, so the line is necessary. It would be a source of error in general, because you will never forget to set the contents of the DF, yet you may not realise that setting the index could even matter.
RAM could be saved only in the case where _computeInputs is not id.

Reply:
To make this "cleaner", we could consider changing the interface of `_predict` to not return a DF but a more low-level result instead and make `predict` construct the DF correctly.

Reply:
I'll open a separate issue which we may or may not resolve some time